### ISSUE TITLE
[Bug]: [ResourceGroup] Search interrupted after transfering the only node to the other resource groups

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#22356]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230221-eac4523c
- Deployment mode(standalone or cluster): cluster with 3 querynodes
- MQ type(rocksmq, pulsar or kafka):  pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):
```

### Current Behavior

Search failed after transferring the only query node to the other rg 

### Expected Behavior

search keeps succ

### Steps To Reproduce

```markdown
1. create a rgB with 1 query node
2. prepare a collectionC with data and loaded it into rgB with 1 replica
3. keep searching on collectionC
4. transfer the query node from rgB to default rg
Expected: search keeps successfully
```

### Milvus Log

node transfer at 2-23 09:11 beijing time:
```
yanliang-cluster-1cu-milvus-datanode-6cf8684b97-kgps9             1/1     Running     0              40h
yanliang-cluster-1cu-milvus-indexnode-5d7b498b6b-7tkr9            1/1     Running     0              40h
yanliang-cluster-1cu-milvus-mixcoord-596b4cc95-nvvd6              1/1     Running     0              40h
yanliang-cluster-1cu-milvus-proxy-59cb4c745-jfxqz                 1/1     Running     0              40h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-bzmcm            1/1     Running     0              40h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-qgl28            1/1     Running     0              40h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-rcw79            1/1     Running     0              40h
```

### Anything else?

at the first second after node transferring:
1. there is outgoing node info in rgB, which i think is correct. 
2. there is no incoming node info in default rg, which i think is wrong.
![image](https://user-images.githubusercontent.com/82361606/220801593-96d6a314-83b9-4e25-a925-23b2c06eb714.png)
But after a few seconds, the outgoing node info is empty, which i think is wrong.
![image](https://user-images.githubusercontent.com/82361606/220801686-fafd9d39-be58-478f-8c9b-f0059e980c4c.png)

**Comment by weiliu1031** [comment]
need querycoord pod name @yanliang567

**Comment by yanliang567** [comment]
its a mixcoord mode

**Comment by yanliang567** [comment]
A good news is that I then transferred the replica of collectionC from rgB to default rg, the search operations turn to success in seconds.

**Comment by weiliu1031** [comment]
fix on #22370

please verify on this

**Comment by yanliang567** [comment]
verified on 2.2.0-20230228-91d251ab
